### PR TITLE
Laboratory work #1, Vladislava Tsvetkova - 22FPL2

### PR DESCRIPTION
**Pull request by Vladays**

### CODE REVIEW

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Oh.. Where is the implementation of the 1st lab? :(

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -11,6 +11,19 @@ def tokenize(text: str) -> list[str] | None:
...
+    text=text.lower()
+    wronglist=(list(text))
+    goodlist=[]
+    for item in wronglist:
+        if item.isalpha():
+            goodlist.append(item)
+    return(goodlist)
+
+
+tokenize("texugyTTRETHFDFG546//t")
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:25]: remove

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -11,6 +11,19 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text,str):
+        return None
+    text=text.lower()
+    wronglist=(list(text))
+    goodlist=[]
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:18]: in one line, please

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -19,6 +32,16 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
+    if not isinstance(tokens,list):
+        return None
+    numberallitems=len(tokens)
+    freq_dict={}
+    for item in tokens:
+        numberi=tokens.count(item)
+        freqi=numberi/numberallitems
+        freq_dict[item]=freqi
+    return(freq_dict)
+calculate_frequencies(['t', 'e', 'x', 'u', 'g', 'y', 't', 't', 'r', 'e', 't', 'h', 'f', 'd', 'f', 'g', 't'])
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:44]: remove

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -19,6 +32,16 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
     """
+    if not isinstance(tokens,list):
+        return None
+    numberallitems=len(tokens)
+    freq_dict={}
+    for item in tokens:
+        numberi=tokens.count(item)
+        freqi=numberi/numberallitems
+        freq_dict[item]=freqi
+    return(freq_dict)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:43]: do you call return?

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -28,6 +51,18 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
...
+        return None
+    if not isinstance(text, str):
+        return None
+    goodlist=tokenize(text)
+    freq_dict=calculate_frequencies(goodlist)
+    profilel={"name": language, "freq": {}}
+
+    for key, value in freq_dict.items():
+        profilel["freq"][(key)]=(value)
+    return(profilel)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:64]: .

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -28,6 +51,18 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
...
+    if not isinstance(language, str):
+        return None
+    if not isinstance(text, str):
+        return None
+    goodlist=tokenize(text)
+    freq_dict=calculate_frequencies(goodlist)
+    profilel={"name": language, "freq": {}}
+
+    for key, value in freq_dict.items():
+        profilel["freq"][(key)]=(value)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:63]: to many breakets

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -37,8 +72,21 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
...
+        return None
+    if len(actual)==0:
+        return None
+    else:
+        dlina=len(actual)
+        for index in range(dlina):
+            numeratory += actual[index]
+            numeratorp += predicted[index]
+            numerator += (numeratory - numeratorp)**2
+    mse=(numerator/dlina)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:89]: strange implementation

File: lab_1_classify_by_unigrams/start.py [hunk 8]
```diff
@@ -16,6 +16,87 @@ def main() -> None:
     result = None
     assert result, "Detection result is None"
 
+    def tokenize(text: str) -> list[str] | None:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/start.py:19]: What? remove

File: seminars/practice_2_string.py [hunk 9]
```diff
@@ -16,8 +16,152 @@
 
 # Create a string
 example = 'Hello'  # or "Hello"
+
```
Comment (artyomtugaryov) [inline seminars/practice_2_string.py:19]: very productive seminar, I guess... Revert it.

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 2 above)
Comment (Vladays) [reply lab_1_classify_by_unigrams/main.py:17]: if you about wronglist - improved

File: lab_1_classify_by_unigrams/main.py [hunk 10]
```diff
@@ -12,10 +10,14 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
-    if not isinstance(text, str):
+    if not isinstance(text,str):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:13]: add space, please

File: lab_1_classify_by_unigrams/main.py [hunk 11]
```diff
@@ -24,13 +26,20 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
-    ):
+    if not isinstance(tokens,list):
         return None
+    for token in tokens:
+        if not isinstance(token,str):
+            return None
+        numberallitems=len(tokens)
+        freq_dict={}
+        for item in tokens:
+            numberi=tokens.count(item)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:37]: spaces

File: seminars/practice_2_string.py
Comment (artyomtugaryov) [inline seminars/practice_2_string.py:1]: revert

File: main.py
Comment (artyomtugaryov) [inline main.py:1]: revert

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
Comment (Vladays) [reply lab_1_classify_by_unigrams/main.py:83]: Does it mean wrong and should be changed?

File: lab_1_classify_by_unigrams/main.py (hunk 10 above)
Comment (Vladays) [reply lab_1_classify_by_unigrams/main.py]: improved

File: lab_1_classify_by_unigrams/main.py (hunk 11 above)
Comment (Vladays) [reply lab_1_classify_by_unigrams/main.py]: improved

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (Vladays) [reply lab_1_classify_by_unigrams/main.py]: removed

File: lab_1_classify_by_unigrams/main.py (hunk 4 above)
Comment (Vladays) [reply lab_1_classify_by_unigrams/main.py]: improved

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (Vladays) [reply lab_1_classify_by_unigrams/main.py]: changed

File: lab_1_classify_by_unigrams/main.py (hunk 6 above)
Comment (Vladays) [reply lab_1_classify_by_unigrams/main.py]: changed

File: lab_1_classify_by_unigrams/start.py (hunk 8 above)
Comment (Vladays) [reply lab_1_classify_by_unigrams/start.py]: removed

File: lab_1_classify_by_unigrams/main.py [hunk 12]
```diff
@@ -120,45 +129,29 @@ def detect_language(
...
-    if profile_1_metric < profile_2_metric:
-        return str(profile_1['name'])
-
-    return [profile_1['name'], profile_2['name']].sort()
-
+    if mse_1>mse_2:
+        return str(profile_2["name"])
+    if mse_1<mse_2:
+        return str(profile_1["name"])
+    return [profile_1["name"], profile_2["name"]].sort()
```
Comment (marina-kaz) [inline lab_1_classify_by_unigrams/main.py:146]: Why do we return a list, not a string?

File: main.py
Comment (artyomtugaryov) [inline main.py:1]: remove

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

**Pull request closed without merging**